packages/contracts-ops/contracts/scripts/DeployImplementations.s.sol

```diff
@@ -0,0 +1,177 @@
+import {Home} from "@nomad-xyz/contracts-core/contracts/Home.sol";
+import {Replica} from "@nomad-xyz/contracts-core/contracts/Replica.sol";
+import {GovernanceRouter} from "@nomad-xyz/contracts-core/contracts/governance/GovernanceRouter.sol";
+import {IUpdaterManager} from "@nomad-xyz/contracts-core/contracts/interfaces/IUpdaterManager.sol";
+// Bridge
+import {BridgeRouter, EthereumBridgeRouter} from "@nomad-xyz/contracts-bridge/contracts/BridgeRouter.sol";
+import {BridgeToken} from "@nomad-xyz/contracts-bridge/contracts/BridgeToken.sol";
+import {TokenRegistry} from "@nomad-xyz/contracts-bridge/contracts/TokenRegistry.sol";
+// Ops
+import {Config} from "../Config.sol";
+import {JsonWriter} from "../JsonWriter.sol";
+// Utilities
+import {Script} from "forge-std/Script.sol";
+
+abstract contract DeployImplementationsLogic is Script, Config {
+    /*//////////////////////////////////////////////////////////////
+                          DEPLOYED CONTRACTS
+    //////////////////////////////////////////////////////////////*/
+
+    Home home;
+    Replica replica;
+    GovernanceRouter governanceRouter;
+    BridgeRouter bridgeRouter;
+    TokenRegistry tokenRegistry;
+    BridgeToken bridgeToken;
+
+    /*//////////////////////////////////////////////////////////////
+                       IMPLEMENTATION DEPLOYMENT
+    //////////////////////////////////////////////////////////////*/
+
+    // NOTE: these init values do not map to correct expected values.
+    // Storage variables in implementation contracts don't matter.
+    // Purpose is to initialize the implementations as a matter of best practice,
+    // despite the fact that in Nomad's architecture,
+    // un-initialized implementations can't harm the protocol
+    // (unless, in the future, we introduce delegatecall in any implementations)
+    function deployImplementations(string memory _domain) internal {
+        // Home
+        home = new Home(getDomainNumber(_domain));
+        home.initialize(IUpdaterManager(address(getUpdaterManager(_domain))));
+        // Replica
+        replica = new Replica(getDomainNumber(_domain));
+        replica.initialize(0, address(0), bytes32(0), 0);
+        // GovernanceRouter
+        governanceRouter = new GovernanceRouter(
+            getDomainNumber(_domain),
+            getRecoveryTimelock(_domain)
+        );
+        governanceRouter.initialize(
+            address(getXAppConnectionManager(_domain)),
+            address(0)
+        );
+        // BridgeRouter
+        if (keccak256(bytes(_domain)) == keccak256(bytes("ethereum"))) {
+            bridgeRouter = BridgeRouter(
+                address(
+                    new EthereumBridgeRouter(address(getAccountant(_domain)))
+                )
+            );
+        } else {
+            bridgeRouter = new BridgeRouter();
+        }
+        BridgeRouter(bridgeRouter).initialize(
+            address(tokenRegistry),
+            address(getXAppConnectionManager(_domain))
```

Comment:
why initialize implemetnation with real data?

Reply:
i believe both of these would revert if you passed `address(0)` because they are cast as contracts in the `initialize` function

Reply:
https://github.com/nomad-xyz/monorepo/pull/527/commits/ee6232b67f40972eefbcb7e7900b131aca5ff145